File: nba-tracker-api/app/schemas/shotchart.py
from typing import List, Optional
from pydantic import BaseModel


class ShotDetail(BaseModel):
    """Individual shot data with X/Y coordinates"""
    grid_type: Optional[str] = None  # Grid type (from API)
    game_id: Optional[str] = None
    game_event_id: Optional[int] = None
    player_id: int
    player_name: str
    team_id: int
    team_name: str
    period: int
    minutes_remaining: int
    seconds_remaining: int
    event_type: Optional[str] = None
    action_type: Optional[str] = None
    shot_type: Optional[str] = None
    shot_zone_basic: Optional[str] = None
    shot_zone_area: Optional[str] = None
    shot_zone_range: Optional[str] = None
    shot_distance: Optional[float] = None
    loc_x: Optional[int] = None  # X coordinate on court
    loc_y: Optional[int] = None  # Y coordinate on court
    shot_attempted_flag: int
    shot_made_flag: int
    game_date: Optional[str] = None
    htm: Optional[str] = None  # Home team abbreviation
    vtm: Optional[str] = None  # Visitor team abbreviation


class LeagueAverage(BaseModel):
    """League average shooting by zone"""
    grid_type: Optional[str] = None
    shot_zone_basic: Optional[str] = None
    shot_zone_area: Optional[str] = None
    shot_zone_range: Optional[str] = None
    fga: Optional[int] = None
    fgm: Optional[int] = None
    fg_pct: Optional[float] = None


class ShotChartResponse(BaseModel):
    """Response containing shot chart data"""
    player_id: int
    player_name: str
    team_id: int
    team_name: str
    season: str
    season_type: str
    shots: List[ShotDetail]
    league_averages: List[LeagueAverage]

File: nba-tracker/src/types/shotchart.ts
export interface ShotDetail {
  game_id?: string;
  game_event_id?: number;
  player_id: number;
  player_name: string;
  team_id: number;
  team_name: string;
  period: number;
  minutes_remaining: number;
  seconds_remaining: number;
  event_type?: string;
  action_type?: string;
  shot_type?: string;
  shot_zone_basic?: string;
  shot_zone_area?: string;
  shot_zone_range?: string;
  shot_distance?: number;
  loc_x?: number; // X coordinate on court
  loc_y?: number; // Y coordinate on court
  shot_attempted_flag: number;
  shot_made_flag: number;
  game_date?: string;
  htm?: string; // Home team abbreviation
  vtm?: string; // Visitor team abbreviation
}

export interface LeagueAverage {
  grid_type?: string;
  shot_zone_basic?: string;
  shot_zone_area?: string;
  shot_zone_range?: string;
  fga?: number;
  fgm?: number;
  fg_pct?: number;
}

export interface ShotChartResponse {
  player_id: number;
  player_name: string;
  team_id: number;
  team_name: string;
  season: string;
  season_type: string;
  shots: ShotDetail[];
  league_averages: LeagueAverage[];
}

